google/cloud/bigtable/row_response.py

```diff
@@ -36,47 +42,237 @@ class RowResponse(Sequence["CellResponse"]):
     cells = row["family", "qualifier"]
     """
 
-    def __init__(self, key: row_key, cells: list[CellResponse]):
+    def __init__(
+        self,
+        key: row_key,
+        cells: list[CellResponse]
+        | dict[tuple[family_id, qualifier], list[dict[str, Any]]],
+    ):
+        """Expected to be used internally only"""
         self.row_key = key
-        self.cells: OrderedDict[
-            family_id, OrderedDict[qualifier, list[CellResponse]]
+        self._cells_map: dict[
+            family_id, dict[qualifier, list[CellResponse]]
         ] = OrderedDict()
-        """Expected to be used internally only"""
-        pass
+        self._cells_list: list[CellResponse] = []
+        if isinstance(cells, dict):
+            # handle dict input
+            tmp_list = []
+            for (family, qualifier), cell_list in cells.items():
+                for cell_dict in cell_list:
+                    cell_obj = CellResponse._from_dict(
+                        key, family, qualifier, cell_dict
+                    )
+                    tmp_list.append(cell_obj)
+            cells = tmp_list
+        # add cells to internal stores using Bigtable native ordering
+        for cell in sorted(cells):
+            if cell.row_key != self.row_key:
+                raise ValueError(
+                    f"CellResponse row_key ({cell.row_key!r}) does not match RowResponse key ({self.row_key!r})"
+                )
+            if cell.family not in self._cells_map:
+                self._cells_map[cell.family] = OrderedDict()
+            if cell.column_qualifier not in self._cells_map[cell.family]:
+                self._cells_map[cell.family][cell.column_qualifier] = []
+            self._cells_map[cell.family][cell.column_qualifier].append(cell)
+            self._cells_list.append(cell)
 
     def get_cells(
-        self, family: str | None, qualifer: str | bytes | None
+        self, family: str | None = None, qualifier: str | bytes | None = None
     ) -> list[CellResponse]:
         """
         Returns cells sorted in Bigtable native order:
             - Family lexicographically ascending
-            - Qualifier lexicographically ascending
+            - Qualifier ascending
```

Comment:
I had in my notes that Qualifier should be sorted lexographically, but I assume this is a mistake, since qualifier is bytes? 
I implemented this by directly comparing the bytes values instead

Reply:
java-bigtable uses a byteStringComaprator: https://github.com/googleapis/java-bigtable/blob/ef7d7419293f9f046dcf941148a86da713feae17/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/internal/ByteStringComparator.java

https://github.com/googleapis/java-bigtable/blob/ef7d7419293f9f046dcf941148a86da713feae17/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/RowCell.java#L51